Jantimizer/QueryOptimiser/IQueryOptimiser.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Histograms.Managers;
+using Histograms;
+using QueryOptimiser.QueryGenerators;
+using QueryParser;
+using DatabaseConnector;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using QueryParser.Models;
+using QueryOptimiser.Cost.CostCalculators;
+using QueryOptimiser.Cost.Nodes;
+
+namespace QueryOptimiser
+{
+    public interface IQueryOptimiser<HistogramType, ConnectorType>
+        where HistogramType : IHistogram
+        where ConnectorType : IDbConnector
+    {
+        public IParserManager ParserManager { get; set; }
```

Comment:
Might be worth it to decouple a bit here. Since the only thing this is used for is an overloaded method.
It might make a little more sense to just remove this and the overloaded method.